### PR TITLE
Plasma Windoors

### LINKED ISSUE #3244: Deconstructing Windoors gives infinite cable coils
On the part where you cut the wires from the windoor, spamming it over and over again will give you a cable piece for each time you click. Putting one cable back into the windoor and spam cutting the wires again will allow you to keep getting like 20 pieces of cable every time you put one piece in.

### PR DESCRIPTION
**Pull request by clusterfack**
-Plasma windoors are constructible, base strengths are 3x and 4x regular windoor strength
-Regular windoor strengths reduced a small amount (1/3) to 100 and 200 health
-Fixes #3244, cable coils cannot be duped from windoors

### CODE REVIEW

**Comment by TermedClepe** [comment]
1/3 is not a small amount.

**Comment by clusterfack** [comment]
People were saying they take way  too long to bash anyway. Now imagine that + plasmafied.

**Comment by d3athrow** [comment]
You should add a way to deconstruct windoors.

**Comment by frangi** [comment]
well, since they require an airlock controller, why not make them behave in that sense like a standard airlock?

**Comment by clusterfack** [comment]
Actually you already can, just crowbar an opened windoor and boom the airlock electronics comes out. I'm not exactly sure when that was put in.

File: code/game/machinery/doors/windowdoor.dm [hunk 1]
```diff
@@ -1,16 +1,17 @@
 /obj/machinery/door/window
-	name = "interior door"
-	desc = "A strong door."
+	name = "Window Door"
+	desc = "A sliding glass door."
 	icon = 'icons/obj/doors/windoor.dmi'
 	icon_state = "left"
 	var/base_state = "left"
-	var/health = 150.0 //If you change this, consiter changing ../door/window/brigdoor/ health at the bottom of this .dm file
+	var/health = 100 //If you change this, consiter changing ../door/window/brigdoor/ health at the bottom of this .dm file
```
Comment (BestName) [inline code/game/machinery/doors/windowdoor.dm]: >consiter

Could you please fix this while you're at it?

File: code/game/machinery/doors/windowdoor.dm [hunk 2]
```diff
@@ -290,19 +297,53 @@
...
+	// Windoor assembly
+	var/obj/structure/windoor_assembly/plasma/WA = new /obj/structure/windoor_assembly/plasma(src.loc)
+	set_assembly(user, WA)
+	return WA
+
+/obj/machinery/door/window/plasma/take_damage(var/damage)
+	src.health = max(0, src.health - damage)
+	if (src.health <= 0)
+		getFromPool(/obj/item/weapon/shard/plasma, loc)
+		var/obj/item/stack/cable_coil/CC = new /obj/item/stack/cable_coil(src.loc)
```
Comment (ComicIronic) [inline code/game/machinery/doors/windowdoor.dm]: This should use pooling.

Also there is an exceptional amount of copied code here and it should really be handled more OOP.
You could make a proc for generating the scrap, for example.

Comment (clusterfack) [inline code/game/machinery/doors/windowdoor.dm]: Yeah that's true that proc does pretty much just copy the other, wouldn't say an 'exceptional' amount of copied code though

**Comment by clusterfack** [comment]
Done

**Comment by d3athrow** [comment]
>Actually you already can, just crowbar an opened windoor and boom the airlock electronics comes out. I'm not exactly sure when that was first made possible.

Sheeit

**Pull request merged**